src/scripts/Build-Bicep.ps1

```diff
@@ -0,0 +1,189 @@
+<#
+.SYNOPSIS
+    Builds all toolkit templates for publishing to Azure Quickstart Templates.
+.DESCRIPTION
+    Run this from the /src/scripts folder.
+.EXAMPLE
+    ./Build-Bicep ../bicep-registry/module-name
+    Generates separate modules for each supported scope for the specified module.
+.EXAMPLE
+    ./Build-Bicep ../bicep-registry/module-name -Scope subscription
+    Generates the module for only one scope (subscription in this case).
+.EXAMPLE
+    ./Build-Bicep ../bicep-registry/module-name -Scope subscription -Debug
+    Renders main module and test bicep code to the console instead of generating files.
+.PARAMETER Module
+    Path to the module to build.
+.PARAMETER Scope
+    Optional. Scope to build. If not specified, all scopes will be built.
+.PARAMETER Debug
+    Optional. Renders main module and test bicep code to the console instead of generating files. Line numbers map to original file.
+#>
+Param (
+    [Parameter(Position = 0)][string] $Module,
+    [string] $Scope
+)
+
+# Use the debug flag from common parameters to determine whether to run in debug mode
+$Debug = $DebugPreference -eq "Continue"
+
+$outdir = "../../release"
+$scopeDirective = '//(\s*@(subscription|resourceGroup|managementGroup|tenant))+';
+$dir = Get-Item $Module;
+
+# List of supported scopes to be updated later
+$script:scopes = @()
+
+# Generates modules for each supported scope
+function Build-Modules([string] $Path, [switch] $CopySupportingFiles) {
+    # Read code from the bicep file
+    $lines = (Get-Content (Join-Path $Module $Path));
+
+    # Get the supported scopes
+    $script:scopes = [regex]::Matches($lines[0], "^$scopeDirective\s*$").Groups[2].Captures `
+    | ForEach-Object { $_.Value } `
+    | Where-Object { $_.ToLower() -eq $Scope.ToLower() -or [string]::IsNullOrWhitespace($Scope) }
+
+    # Loop thru each scope
+    $script:scopes `
+    | ForEach-Object { 
+        $currentScope = $_
+        $moduleName = "$($currentScope.ToLower())-$($dir.Name)"
+  
+        if ($CopySupportingFiles) {
+            Write-Host "  $currentScope..."
+        }
+
+        # Init string builder for the scope output
+        $sb = [System.Text.StringBuilder]::new()
+  
+        # Write debug header
+        if ($Debug) {
+            Write-Host $moduleName
+            Write-Host "".PadLeft($moduleName.Length, "=")
+        }
+  
+        # Start on second line to get past the supported scope directive
```

Comment:
This seems a brittle assumption, we should eventually be more flexible so if there is a /n on first line and second line is the scope it still works.

Reply:
Yeah, I thought about that. I was trying to keep it simple :-P I suspect we will need to expand it in the future. A couple other options:
1. Auto-detect supported scopes from the entire file.
2. Look up supported scopes on the targetScope line only:
   ```bicep
   targetScope: 'subscription' // @resourceGroup, @tenant
   ```
Both of those might be better than requiring the first line to define scopes.

Reply:
Changed it to infer scopes and added guidance to add a scope directive with all scopes on the targetScope line as a fallback.